cider/exec/plan/parser/SubstraitToAnalyzerExpr.cpp

```diff
@@ -970,6 +970,36 @@ std::shared_ptr<Analyzer::Expr> Substrait2AnalyzerExprConverter::toAnalyzerExpr(
     // use a int32/int64 to represent under substrait protocol
     if (!s_cast_expr.input().literal().fixed_char().empty()) {
       v.bigintval = dateToInt64(s_cast_expr.input().literal().fixed_char());
+    }
+    // TODO: spevenhe.
+    // support nested function like cast(cast(xx as String) AS Date).
+    // CAST(SUBSTRING(col_string ,0,5) AS DATE
+    // Now only supports cast(col_string AS Date)
+    else if (s_cast_expr.input().selection().has_direct_reference() &&
```

Comment:
can you refactor `selection().has_direct_reference()` and `selection().has_root_reference()` as a new function, I guess  this is checking whether expr is a column.

Reply:
Yes, it is. Updated